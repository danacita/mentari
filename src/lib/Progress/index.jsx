/* Background https://codepen.io/bbrady/pen/ozrjKE */
import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import Color from '../Styles/bases/Color';

const propTypes = {
  size: PropTypes.number.isRequired,
  percent: PropTypes.number.isRequired,
  text: PropTypes.number.isRequired
};

const Progress = ({ size, text, percent }) => {
  const radius = (size - 4) / 2;
  const dashArray = radius * Math.PI * 2;
  const dashOffset = dashArray - (dashArray * percent) / 100;
  return (
    <Fragment>
      <svg width={size} height={size} viewBox={`0 0 ${size} ${size}`}>
        <circle
          cx={size / 2}
          cy={size / 2}
          r={radius}
          strokeWidth={4}
          stroke={Color.lightBlue}
          fill="white"
        />
        <circle
          cx={size / 2}
          cy={size / 2}
          r={radius}
          strokeWidth={4}
          transform={`rotate(-90 ${size / 2} ${size / 2})`}
          stroke={Color.orange}
          fill="transparent"
          style={{
            strokeDasharray: dashArray,
            strokeDashoffset: dashOffset,
            strokeLinecap: 'round',
            strokeLinejoin: 'round'
          }}
        />
        <text
          style={{
            fontWeight: 600,
            fill: Color.darkBlue
          }}
          x="50%"
          y="50%"
          dy=".3em"
          textAnchor="middle"
        >
          {text}
        </text>
      </svg>
    </Fragment>
  );
};

Progress.propTypes = propTypes;

export default Progress;
