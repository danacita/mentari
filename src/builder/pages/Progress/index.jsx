import React, { Component } from 'react';
import SyntaxHighlighter from 'react-syntax-highlighter';
import { atomOneDarkReasonable } from 'react-syntax-highlighter/dist/esm/styles/hljs';
import { ContentWithRightBar } from '../Styled';
import { Margin, Progress, Select } from '../../../lib';
import PropsShared from '../../shared/Props';
import DemoShared from '../../shared/Demo';
import CodeShared from '../../shared/Code';
import RightbarShared from '../../shared/Rightbar';

class ProgressPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      handleProgressSelectValue: { id: '30', name: '30' },
      handleProgressSize: { id: '50', name: '50' }
    };
  }

  render() {
    const { handleProgressSelectValue, handleProgressSize } = this.state;
    const codeString = `import { Progress } from '@danacita/mentari';\n<Progress text={${
      handleProgressSelectValue.id
    }} percent={${handleProgressSelectValue.id}} size={${handleProgressSize.id}} />`;
    return (
      <ContentWithRightBar>
        <h1>Progress</h1>
        <p>A progress bar can be used to show a user how far along he/she is in a process.</p>
        <Margin top={24}>
          <PropsShared>
            <li>
              <span>text</span>
              <span>Show a percentage number</span>
            </li>
            <li>
              <span>percent</span>
              <span>Apply a percentage circle active</span>
            </li>
            <li>
              <span>size</span>
              <span>Set size of Progress circle</span>
            </li>
          </PropsShared>
        </Margin>
        <DemoShared>
          <h4>Demo</h4>
          <p>You can change the props via the component builder on your right</p>
          <Margin top={16}>
            <Progress
              text={handleProgressSelectValue.id ? Number(handleProgressSelectValue.id) : 0}
              percent={handleProgressSelectValue.id ? Number(handleProgressSelectValue.id) : 0}
              size={handleProgressSize.id ? Number(handleProgressSize.id) : 50}
            />
          </Margin>
        </DemoShared>
        <h4>Code</h4>
        <Margin top={16}>
          <CodeShared copyCode={codeString}>
            <SyntaxHighlighter language="jsx" style={atomOneDarkReasonable} showLineNumbers>
              {codeString}
            </SyntaxHighlighter>
          </CodeShared>
        </Margin>
        <RightbarShared title="Progress">
          <Margin top={24} bottom={8}>
            <Select
              label="Choose percent active"
              placeholder="Choose percent active"
              withRedux={false}
              fieldData={[
                {
                  id: '10',
                  name: '10'
                },
                {
                  id: '20',
                  name: '20'
                },
                {
                  id: '30',
                  name: '30'
                },
                {
                  id: '40',
                  name: '40'
                },
                {
                  id: '50',
                  name: '50'
                },
                {
                  id: '60',
                  name: '60'
                },
                {
                  id: '70',
                  name: '70'
                },
                {
                  id: '80',
                  name: '80'
                },
                {
                  id: '90',
                  name: '90'
                },
                {
                  id: '100',
                  name: '100'
                }
              ]}
              valueSelect={handleProgressSelectValue}
              onChange={value => this.setState({ handleProgressSelectValue: value })}
            />
          </Margin>
          <Margin top={24} bottom={8}>
            <Select
              label="Choose size progress"
              placeholder="Choose size progress"
              withRedux={false}
              fieldData={[
                {
                  id: '50',
                  name: '50'
                },
                {
                  id: '60',
                  name: '60'
                },
                {
                  id: '70',
                  name: '70'
                },
                {
                  id: '80',
                  name: '80'
                },
                {
                  id: '90',
                  name: '90'
                },
                {
                  id: '100',
                  name: '100'
                }
              ]}
              valueSelect={handleProgressSize}
              onChange={value => this.setState({ handleProgressSize: value })}
            />
          </Margin>
        </RightbarShared>
      </ContentWithRightBar>
    );
  }
}

export default ProgressPage;
