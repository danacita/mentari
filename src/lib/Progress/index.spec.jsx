import React from 'react';
import renderer from 'react-test-renderer';

import Progress from '.';

describe('Progress', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Progress size={10} text={1} percent={25} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
