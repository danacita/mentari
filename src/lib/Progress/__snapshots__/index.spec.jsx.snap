// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders correctly 1`] = `
<svg
  height={10}
  viewBox="0 0 10 10"
  width={10}
>
  <circle
    cx={5}
    cy={5}
    fill="white"
    r={3}
    stroke="#dce9fa"
    strokeWidth={4}
  />
  <circle
    cx={5}
    cy={5}
    fill="transparent"
    r={3}
    stroke="#fcac42"
    strokeWidth={4}
    style={
      Object {
        "strokeDasharray": 18.84955592153876,
        "strokeDashoffset": 14.137166941154069,
        "strokeLinecap": "round",
        "strokeLinejoin": "round",
      }
    }
    transform="rotate(-90 5 5)"
  />
  <text
    dy=".3em"
    style={
      Object {
        "fill": "#3151aa",
        "fontWeight": 600,
      }
    }
    textAnchor="middle"
    x="50%"
    y="50%"
  >
    1
  </text>
</svg>
`;
